draft-ietf-quic-transport.md

```diff
@@ -3847,13 +3875,41 @@ language from Section 3 of {{!TLS13=RFC8446}}.
 {: #figure-transport-parameters title="Definition of TransportParameters"}
 
 The `extension_data` field of the quic_transport_parameters extension defined in
-{{QUIC-TLS}} contains a TransportParameters value.  TLS encoding rules are
-therefore used to describe the encoding of transport parameters.
+{{QUIC-TLS}} contains a TransportParameters value.
 
 QUIC encodes transport parameters into a sequence of octets, which are then
 included in the cryptographic handshake.
 
 
+## Version Validation Fields {#transport-parameter-versions}
+
+Client and server exchange information about versions along with transport
+parameters in order to protect against version downgrade attacks (see
+{{version-validation}}).
+
+The `supported_versions` field is a list of the versions the client supports.
+This field enables upgrade to compatible versions (see {{version-upgrade}}).  A
+client MAY omit versions from this list if they appear in the
+`unsupported_versions` field.
+
+The `unsupported_versions` field is a list of the versions the client supports,
+but which the client believes that the server does not support.  This list MUST
+be empty unless the client has received and processed a Version Negotiation
+packet.  A client MUST NOT omit a version that it supports from
+`unsupported_versions`, unless the Version Negotiation packet it processed
+listed that version.
```

Comment:
I wonder if this _unless_ clause contradicts with the following sentence added above: _Though different versions might convey information about versions differently, a client MUST NOT change the set of versions it claims to support._

Reply:
Hmm, the text isn't *wrong*, but I'm not sure how to fix it.  The set of versions that a client claims to support is the union of `supported_versions` and `unsupported_versions`.  I've said that, but I think that there's a better way to structure the text to avoid having to say that.